ann_benchmarks/algorithms/constructors.py

```diff
@@ -0,0 +1,63 @@
+from __future__ import absolute_import
+import sys
+import importlib
+import traceback
+
+__constructors = [
```

Comment:
could probably just import these rather than using strings?

Reply:
(not blocking for PR)

Reply:
This code has two goals:

* we want to expose only some explicitly enumerated constructors through the configuration file; and
* we want to be able to distinguish between a constructor that's not been installed -- in which case we should skip its experiments --  and one that's not known to the framework (which should be treated as a hard error).

I suppose this *could* also be done by putting every constructor symbol in a separate `import` statement wrapped up in its own `try`/`except ImportError` block, but... I don't know, that seems messier to me.